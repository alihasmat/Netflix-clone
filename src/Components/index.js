export { default as Accordian } from "./Accordian/Accordian";
export { default as Card } from "./Card/Card";
export { default as Feature } from "./Feature/Feature";
export { default as Footer } from "./Footer/Footer";
export { default as Form } from "./Form/Form";
export { default as Header } from "./Header/Header";
export { default as Jumbotron } from "./Jumbotron/Jumbotron";
export { default as Loading } from "./Loading/Loading";
export { default as Player } from "./Player/Player";
export { default as Profiles } from "./Profiles/Profiles";
export { default as Search } from "./Search/Search";
