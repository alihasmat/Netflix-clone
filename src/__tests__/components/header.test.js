import { fireEvent, render } from "@testing-library/react";
import { Header } from "../../Components";

jest.mock("react-router-dom");

describe("<Header />", () => {
  it("renders basic <Header /> with background", () => {
    const { container, getByText, getByTestId } = render(
      <Header>
        <Header.Frame>
          <Header.Logo src="/Images/misc/logo.svg" alt="logo"></Header.Logo>
          <Header.TextLink active={true}>Hello I am a link.</Header.TextLink>
        </Header.Frame>
      </Header>
    );
    expect(getByText("Hello I am a link.")).toBeTruthy();
    expect(getByTestId("header-bg")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders basic <Header /> without background", () => {
    const { container, getByText, queryByTestId } = render(
      <Header bg={false}>
        <Header.Frame>
          <Header.Logo src="/Images/misc/logo.svg" alt="logo"></Header.Logo>
          <Header.ButtonLink>Sign In</Header.ButtonLink>
          <Header.TextLink active={false}>Hello I am a link.</Header.TextLink>
        </Header.Frame>
      </Header>
    );
    expect(getByText("Hello I am a link.")).toBeTruthy();
    expect(queryByTestId("header-bg")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it(`renders <Header /> page test`, () => {
    const { container, getByText, getByTestId } = render(
      <Header src="joker1" dontShowOnSmallViewPort>
        <Header.Frame>
          <Header.Group>
            <Header.Logo src="/Images/misc/logo.svg" alt="logo" />
            <Header.TextLink active={false} onClick={() => {}}>
              Series
            </Header.TextLink>
            <Header.TextLink active={false} onClick={() => {}}>
              Films
            </Header.TextLink>
          </Header.Group>

          <Header.Group>
            <Header.Search>
              <Header.SearchIcon onClick={() => {}}>
                <img src="images/icons/search.png" />
              </Header.SearchIcon>
              <Header.SearchInput
                value="Joker"
                placeholder="Search..."
                onChange={() => {}}
              />
            </Header.Search>

            <Header.Profile>
              <Header.Picture src="/images/Hasmat.png" />
              <Header.Dropdown>
                <Header.Group>
                  <Header.Picture src="/images/Hasmat.png" />
                  <Header.TextLink active={false}>Hasmat Ali</Header.TextLink>
                </Header.Group>
                <Header.Group>
                  <Header.TextLink active={false} onClick={() => {}}>
                    Sign out
                  </Header.TextLink>
                </Header.Group>
              </Header.Dropdown>
            </Header.Profile>
          </Header.Group>
        </Header.Frame>

        <Header.Feature>
          <Header.FeatureCallOut>Watch Joker Now</Header.FeatureCallOut>
          <Header.Text>Forever alone in a crowd..</Header.Text>
          <Header.PlayButton>Play</Header.PlayButton>
        </Header.Feature>

        <Header.Gradient />
      </Header>
    );
    expect(getByText("Series")).toBeTruthy();
    expect(getByText("Films")).toBeTruthy();

    expect(getByTestId("search-input")).toBeTruthy();
    expect(getByTestId("search-input").value).toBe("Joker");
    fireEvent.change(getByTestId("search-input"), {
      target: { value: "Simpsons" },
    });
    fireEvent.click(getByTestId("search-click"));

    expect(getByText("Hasmat Ali")).toBeTruthy();
    expect(getByText("Sign out")).toBeTruthy();
    expect(getByText("Watch Joker Now")).toBeTruthy();
    expect(getByText("Forever alone in a crowd..")).toBeTruthy();
    expect(getByText("Play")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
