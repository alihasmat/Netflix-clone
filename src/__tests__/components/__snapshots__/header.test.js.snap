// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders <Header /> page test 1`] = `
<div
  class="style__Background-sc-1jm96lg-0 iuoPHl"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="style__Frame-sc-1jm96lg-2 dtmKAa"
  >
    <div
      class="style__Group-sc-1jm96lg-8 bvisSP"
    >
      <p
        class="style__TextLink-sc-1jm96lg-7 hLavQF"
      >
        Series
      </p>
      <p
        class="style__TextLink-sc-1jm96lg-7 hLavQF"
      >
        Films
      </p>
    </div>
    <div
      class="style__Group-sc-1jm96lg-8 bvisSP"
    >
      <div
        class="style__Search-sc-1jm96lg-14 buBCJb"
      >
        <button
          class="style__SearchIcon-sc-1jm96lg-15 hpydqn"
          data-testid="search-click"
        >
          <img
            src="images/icons/search.png"
          />
        </button>
        <input
          class="style__SearchInput-sc-1jm96lg-16 dSyDya"
          data-testid="search-input"
          placeholder="Search..."
          value="Joker"
        />
      </div>
      <div
        class="style__Profile-sc-1jm96lg-11 fARoIl"
      >
        <img
          class="style__Picture-sc-1jm96lg-9 uPQHn"
          src="/images/users//images/Hasmat.png.png"
        />
        <div
          class="style__Dropdown-sc-1jm96lg-10 dlLTgU"
        >
          <div
            class="style__Group-sc-1jm96lg-8 bvisSP"
          >
            <img
              class="style__Picture-sc-1jm96lg-9 uPQHn"
              src="/images/users//images/Hasmat.png.png"
            />
            <p
              class="style__TextLink-sc-1jm96lg-7 hLavQF"
            >
              Hasmat Ali
            </p>
          </div>
          <div
            class="style__Group-sc-1jm96lg-8 bvisSP"
          >
            <p
              class="style__TextLink-sc-1jm96lg-7 hLavQF"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="style__Frame-sc-1jm96lg-2 style__Feature-sc-1jm96lg-5 dtmKAa OxDmk"
  >
    <h2
      class="style__FeatureCallOut-sc-1jm96lg-12 fROkPz"
    >
      Watch Joker Now
    </h2>
    <p
      class="style__Text-sc-1jm96lg-6 vFbeT"
    >
      Forever alone in a crowd..
    </p>
    <button
      class="style__PlayButton-sc-1jm96lg-13 kQpieA"
    >
      Play
    </button>
  </div>
  <div
    class="style__Gradient-sc-1jm96lg-1 bQsqVP"
  />
</div>
`;

exports[`<Header /> renders basic <Header /> with background 1`] = `
<div
  class="style__Background-sc-1jm96lg-0 PeLls"
  data-testid="header-bg"
>
  <div
    class="style__Frame-sc-1jm96lg-2 dtmKAa"
  >
    <p
      class="style__TextLink-sc-1jm96lg-7 hLavQF"
    >
      Hello I am a link.
    </p>
  </div>
</div>
`;

exports[`<Header /> renders basic <Header /> without background 1`] = `
<div
  class="style__Frame-sc-1jm96lg-2 dtmKAa"
>
  <p
    class="style__TextLink-sc-1jm96lg-7 hLavQF"
  >
    Hello I am a link.
  </p>
</div>
`;
